aws/network/security_groups.tf

```diff
@@ -0,0 +1,233 @@
+    Terraform             = true
+  }
+}
+
+resource "aws_security_group" "forms" {
+  name        = "forms"
+  description = "Ingress - Forms"
+  vpc_id      = aws_vpc.forms.id
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_security_group_rule" "forms_ingress_alb" {
+  description              = "Security group rule for Forms Ingress ALB"
+  type                     = "ingress"
+  from_port                = 3000
+  to_port                  = 3000
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.forms.id
+  source_security_group_id = aws_security_group.forms_load_balancer.id
+}
+
+resource "aws_security_group_rule" "forms_egress_privatelink" {
+  description              = "Security group rule for Forms egress through privatelink"
+  type                     = "egress"
+  from_port                = 443
+  to_port                  = 443
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.forms.id
+  source_security_group_id = aws_security_group.privatelink.id
+}
+
+resource "aws_security_group_rule" "forms_egress_database" {
+  description              = "Security group rule for Forms DB egress to Database"
+  type                     = "egress"
+  from_port                = 5432
+  to_port                  = 5432
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.forms.id
+  source_security_group_id = aws_security_group.forms_database.id
+}
+
+resource "aws_security_group_rule" "forms_egress_redis" {
+  description              = "Security group rule for Forms DB egress to Redis"
+  type                     = "egress"
+  from_port                = 6379
+  to_port                  = 6379
+  protocol                 = "tcp"
+  security_group_id        = aws_security_group.forms.id
+  source_security_group_id = aws_security_group.forms_redis.id
+}
+
+resource "aws_security_group" "forms_load_balancer" {
+  name        = "forms-load-balancer"
+  description = "Ingress - forms Load Balancer"
+  vpc_id      = aws_vpc.forms.id
+
+  ingress {
+    protocol    = "tcp"
+    from_port   = 443
+    to_port     = 443
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  ingress {
+    protocol    = "tcp"
+    from_port   = 80
+    to_port     = 80
+    cidr_blocks = ["0.0.0.0/0"]
+  }
+
+  egress {
+    protocol    = "tcp"
+    from_port   = 3000
+    to_port     = 3000
+    cidr_blocks = [var.vpc_cidr_block]
+  }
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_security_group" "forms_egress" {
+  name        = "egress-anywhere"
+  description = "Egress - Forms External Services"
+  vpc_id      = aws_vpc.forms.id
+
+  tags = {
+    (var.billing_tag_key) = var.billing_tag_value
+    Terraform             = true
+  }
+}
+
+resource "aws_security_group_rule" "forms_egress_notify" {
```

Comment:
We could probably rename this rule to external_auth or something along those lines as this rule isn't used for Notify (legacy carryover from Covid Portal)

Reply:
Makes sense - I'll update the name.